xdsl/transforms/stencil_shape_minimize.py

```diff
@@ -42,6 +50,24 @@ def match_and_rewrite(self, op: func.FuncOp, rewriter: PatternRewriter, /):
             op.update_function_type()
 
 
+@dataclass
+class RestrictStoreOp(RewritePattern):
+    restrict: tuple[int, ...] | None = None
+
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: stencil.StoreOp, rewriter: PatternRewriter, /):
+        if self.restrict:
+            new_bounds: list[tuple[int, int]] = []
+            for lower_bound, upper_bound, bound_lim in zip(
+                op.bounds.lb, op.bounds.ub, self.restrict
+            ):
+                new_bounds.append(
+                    (min(lower_bound, bound_lim), min(upper_bound, bound_lim))
```

Comment:
This looks really good. It could probably be a simple and beautiful tuple-comprehension.

Reply:
thanks! I'll attempt to do this